### PR TITLE
Port flow tests to the updated tests framework

### PR DESCRIPTION
**Pull request by shariarriday**
Relates https://github.com/actor-framework/actor-framework/issues/1646

Updates the tests for following files

- mixed
- multicaster
- observe_on
- publish
- single
- for_each
- flat_map
- concat_map
- generation

### CODE REVIEW

File: libcaf_core/caf/flow/for_each.test.cpp [hunk 1]
```diff
@@ -58,29 +46,31 @@ SCENARIO("for_each iterates all values in a stream") {
...
           auto outputs = std::vector<int>{};
-          ctx->make_observable()
+          make_observable()
             .repeat(7) //
             .as_observable()
             .take(7)
             .map([](int x) { return x * 3; })
-            .do_on_error([](const error& err) { FAIL("on_error: " << err); })
+            .do_on_error([this](const error& err) {
+              fail(std::string_view("on_error: " + caf::to_string(err)));
```
Comment (Neverlord) [inline libcaf_core/caf/flow/for_each.test.cpp:67]:
```suggestion
              fail("on_error: " + caf::to_string(err));
```

File: libcaf_core/caf/flow/multicaster.test.cpp [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        auto print_stats = [&] {
+          const auto& states = uut.impl().observers();
+          for (size_t index = 0; index < states.size(); ++index) {
+            printf("observer %d: buf size = %d; demand = %d\n", (int) index,
+                   (int) states[index]->buf.size(),
+                   (int) states[index]->demand);
+          }
+        };
+        print_stats();
+
```
Comment (Neverlord) [inline libcaf_core/caf/flow/multicaster.test.cpp:48]: Either drop or use `print_debug`.

```suggestion
```

**Review by Neverlord — CHANGES_REQUESTED** [review]

File: libcaf_core/caf/flow/for_each.test.cpp (hunk 1 above)
Comment (shariarriday) [reply libcaf_core/caf/flow/for_each.test.cpp:67]: @Neverlord We can not use `fail("on_error: " + caf::to_string(err));` as we will face the following error `cannot convert from 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>' to 'caf::format_string_with_location'`.

Comment (Neverlord) [reply libcaf_core/caf/flow/for_each.test.cpp:67]: Right, this takes a format string. Then there's no need at all to build a string in the first place: `fail("on_error: {}", caf::to_string(err))`. 🙂

**Review by Neverlord — APPROVED** [review]

**Pull request merged**